packages/aws-cdk/lib/api/hotswap/ecs-services.ts

```diff
@@ -80,6 +85,23 @@ export async function isHotswappableEcsServiceChange(
             },
           },
         });
+
+        if (taskDefinitionResource.hotswappableWithMerge) {
+          // get the latest task definition of the family
+          const describeTaskDefinitionResponse = await sdk
+            .ecs()
+            .describeTaskDefinition({
```

Comment:
No other hotswappable resource does a `describe` on the service. What is missing from `EvaluateCfnTemplate` that forces us to do this? We already have these prior definitions from the previous form of the CFN template.

Reply:
Hi @comcalvi, thanks for looking at this!

> I'm curious what you mean by this. EvaluteCfnTemplate doesn't explicitly check any of these resources, it instead looks up every arg to GetAtt in the current deployed template.

We can't know the actual values for a GetAtt just by looking up the template. For example, given this string: `"Fn::GetAtt": ["SecurityGroupC96ED6A7", "GroupId"]`, how do we know that it should resolve to the security group id? In the current implementation, we have a manually created mapping table for the `GetAtt` functions, and  it can only evaluate one of the resource attributes defined here:

https://github.com/aws/aws-cdk/blob/5ccc56975c323ea19fd0917def51184e13f440d9/packages/aws-cdk/lib/api/evaluate-cloudformation-template.ts#L352-L365 

If we want to evaluate all attributes, we have to manually define these mappings one by one, for all resources and attributes, which is hardly a realistic solution. Since this problem makes ECS hotswap almost unusable for many real use cases, I believe it's worth optimizing the hotswap logic specifically for ECS.

> So instead of going to CFN to get the values, which is what EvaluteCfnTemplate does, we go to the ECS service directly. Why? Is there something that EvaluateCfnTemplate can't handle?

Yes, `EvaluateCfnTemplate` can't handle certain intrinsics, such as `GetAtt` for the attributes that don't have the mappings described above, or `Fn::ImportValue`, etc. An easy way to get the actual values for the attributes is to query the ECS service, which is done by this PR.